File: CS/WebSolution2.Module.Web/DefaultTemplateHelper.cs
using System;
using System.Collections.Generic;
using System.Text;
using DevExpress.ExpressApp.Templates;
using DevExpress.ExpressApp;
using System.Web.Security;
using System.Web;
using DevExpress.ExpressApp.Web;
using System.Web.UI;
using DevExpress.Xpo;
using DevExpress.ExpressApp.Web.Templates;

namespace WebSolution2.Module.Web {
    public class DefaultTemplateHelper {
        private const string PreferredTemplateKey = "PreferredTemplate";
        public static void LoadTemplateType() {
            HttpCookie userTemplateCookie = HttpContext.Current.Request.Cookies[PreferredTemplateCookieName];
            if (userTemplateCookie != null && !String.IsNullOrEmpty(userTemplateCookie.Value)) {
                WebApplication.PreferredApplicationWindowTemplateType = (TemplateType)Enum.Parse(typeof(TemplateType), userTemplateCookie.Value);
            }
        }
        public static void SetTemplateType(TemplateType templateType) {
            WebApplication.PreferredApplicationWindowTemplateType = templateType;
            HttpCookie userTemplateCookie = HttpContext.Current.Response.Cookies[PreferredTemplateCookieName];
            userTemplateCookie.Value = templateType.ToString();
            userTemplateCookie.Expires = DateTime.Now.AddMonths(1);
            HttpContext.Current.Response.SetCookie(userTemplateCookie);
        }
        public static TemplateType GetTemplateType() {
            return WebApplication.PreferredApplicationWindowTemplateType;
        }
        private static string PreferredTemplateCookieName {
            get {
                return PreferredTemplateKey + SecuritySystem.CurrentUserId.ToString();
            }
        }
    }
}

File: VB/WebSolution2.Module.Web/DefaultTemplateHelper.vb
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports DevExpress.ExpressApp.Templates
Imports DevExpress.ExpressApp
Imports System.Web.Security
Imports System.Web
Imports DevExpress.ExpressApp.Web
Imports System.Web.UI
Imports DevExpress.Xpo
Imports DevExpress.ExpressApp.Web.Templates

Namespace WebSolution2.Module.Web
    Public Class DefaultTemplateHelper
        Private Const PreferredTemplateKey As String = "PreferredTemplate"
        Public Shared Sub LoadTemplateType()
            Dim userTemplateCookie As HttpCookie = HttpContext.Current.Request.Cookies(PreferredTemplateCookieName)
            If userTemplateCookie IsNot Nothing AndAlso (Not String.IsNullOrEmpty(userTemplateCookie.Value)) Then
                WebApplication.PreferredApplicationWindowTemplateType = DirectCast(System.Enum.Parse(GetType(TemplateType), userTemplateCookie.Value), TemplateType)
            End If
        End Sub
        Public Shared Sub SetTemplateType(ByVal templateType As TemplateType)
            WebApplication.PreferredApplicationWindowTemplateType = templateType
            Dim userTemplateCookie As HttpCookie = HttpContext.Current.Response.Cookies(PreferredTemplateCookieName)
            userTemplateCookie.Value = templateType.ToString()
            userTemplateCookie.Expires = Date.Now.AddMonths(1)
            HttpContext.Current.Response.SetCookie(userTemplateCookie)
        End Sub
        Public Shared Function GetTemplateType() As TemplateType
            Return WebApplication.PreferredApplicationWindowTemplateType
        End Function
        Private Shared ReadOnly Property PreferredTemplateCookieName() As String
            Get
                Return PreferredTemplateKey + SecuritySystem.CurrentUserId.ToString()
            End Get
        End Property
    End Class
End Namespace